apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/CustomScalar.kt

```diff
@@ -3,15 +3,17 @@ package com.apollographql.apollo.api
 /**
  * Represents a mapping from a custom GraphQL scalar type to a Java/Kotlin class
  */
-interface ScalarType {
-
+data class CustomScalar(
   /**
    * GraphQL schema custom scalar type name (e.g. `ID`, `URL`, `DateTime` etc.)
    */
-  val graphqlName: String
+  val graphqlName: String,
 
   /**
    * Fully qualified class name this GraphQL scalar type is mapped to (e.g. `java.lang.String`, `java.net.URL`, `java.util.DateTime`)
    */
   val className: String
+) {
+  // Do not remove, this is used by generated code
+  companion object
```

Comment:
where it's used? I might missed that during reviewing PR

Reply:
My bad, it's a leftover from a previous attempt to generate CustomScalars as extension properties like `CustomScalar.Date`. But it leads to nasty import clashes (extension property vs `java.util.Date` for an exemple) so I reverted to using `CustomScalars.Date` (See https://github.com/apollographql/apollo-android/pull/2820/commits/da7da119a3d58144aa9225955e9e7a4268b65b95)